test/unit/test_configcommon.py

```
@@ -152,4 +153,19 @@ def test_yaml_deprecation_warning(tmp_path, autosubmit_config: Callable):
 
     new_yaml_file = Path(ini_file.parent,ini_file.stem).with_suffix('.yml')
     assert new_yaml_file.exists()
-    assert new_yaml_file.stat().st_size > 0
\ No newline at end of file
+    assert new_yaml_file.stat().st_size > 0
+
+
+def test_key_error_raise(autosubmit_config: Callable):
+    """Test that a KeyError is raised when a key is not found in the configuration."""
+    as_conf: AutosubmitConfig = autosubmit_config(expid="a000", experiment_data={})
+    with pytest.raises(AutosubmitCritical):
+        as_conf.jobs_data
+
+    with pytest.raises(AutosubmitCritical):
+        as_conf.platforms_data
```

Can this line be inside the previous `with` statement, instead of having two different `with pytest.raises(AutosubmitCritical):?`

---

Not really, both properties should raise an exception. If it is in one with statement, the test will pass if only one raise exception but we need to be sure that both do it.

---

True, thanks! Nothing else from my side then